### PR TITLE
Issue237: Add type to AlphaNodes for introspection

### PR DESCRIPTION
**Pull request by EthanEChristian**
A quick addition for #237 

@mrrodriguez  @WilliamParker

### CODE REVIEW

File: src/main/clojure/clara/rules/engine.cljc [hunk 1]
```diff
@@ -524,7 +524,7 @@
 ;; Record representing alpha nodes in the Rete network,
 ;; each of which evaluates a single condition and
 ;; propagates matches to its children.
-(defrecord AlphaNode [id env children activation]
+(defrecord AlphaNode [id env children activation type]
```
Comment (mrrodriguez) [inline src/main/clojure/clara/rules/engine.cljc]: So the id part of #237 has already been done?

Also, should it perhaps be `fact-type` to be clearer?

Comment (EthanEChristian) [reply src/main/clojure/clara/rules/engine.cljc]: yes, and i have updated it to be fact-type

**Comment by EthanEChristian** [comment]
@mrrodriguez,
For the Durability question, I would expect that any type provided would already have to be serializable as we have to serialize the alpha-roots, ie.
```
{fact-type [AlphaNode]}
``` 

It wouldn't be unreasonable to believe that a user could provide a fact-type-fn that produced something that wasn't serializable, but that would have tanked durability before these changes as well. Additionally, i would think that based on the contract(fressian-handlers), that in the event that a fact-type-fn like this was provided then the user should also provide the appropriate handlers.

**Comment by mrrodriguez** [comment]
https://github.com/cerner/clara-rules/pull/450#issuecomment-582134204

@EthanEChristian that makes sense to me. I wasn't positive we were already serializing this same value - but indeed that should be the same thing.

**Comment by mrrodriguez** [comment]
When this is done, does #237 need closed?

**Comment by EthanEChristian** [comment]
Yes, once this goes in i was going to close it.

I was trying to knock this out in an attempt to get better generated function names for #261 and #291

**Comment by EthanEChristian** [comment]
Alright, I am going to merge this and get #237 closed out

**Pull request merged**